core/src/main/java/org/apache/iceberg/stats/StatsFormat.java

```diff
@@ -0,0 +1,162 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.stats;
+
+import io.airlift.compress.Compressor;
+import io.airlift.compress.zstd.ZstdCompressor;
+import io.airlift.compress.zstd.ZstdDecompressor;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import javax.annotation.Nullable;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableTable;
+import org.apache.iceberg.relocated.com.google.common.collect.Table;
+import org.apache.iceberg.util.ByteBuffers;
+
+public final class StatsFormat {
+  private StatsFormat() {
+  }
+
+  enum Flag {
+    FOOTER_PAYLOAD_COMPRESSED(0, 0),
+    /**/;
+
+    private static final Table<Integer, Integer, Flag> byByteAndBit = Stream.of(values())
+        .collect(ImmutableTable.toImmutableTable(Flag::byteNumber, Flag::bitNumber, Function.identity()));
+
+    private final int byteNumber;
+    private final int bitNumber;
+
+    Flag(int byteNumber, int bitNumber) {
+      this.byteNumber = byteNumber;
+      this.bitNumber = bitNumber;
+    }
+
+    @Nullable
+    static Flag fromBit(int byteNumber, int bitNumber) {
+      return byByteAndBit.get(byteNumber, bitNumber);
+    }
+
+    public int byteNumber() {
+      return byteNumber;
+    }
+
+    public int bitNumber() {
+      return bitNumber;
+    }
+  }
+
+  static final int FOOTER_MAGIC_OFFSET = 0;
+  static final int FOOTER_STRUCT_LENGTH = 12;
+  static final int FOOTER_STRUCT_PAYLOAD_SIZE_OFFSET = 0;
+  static final int FOOTER_STRUCT_FLAGS_OFFSET = 4;
+  static final int FOOTER_STRUCT_FLAGS_LENGTH = 4;
+  static final int FOOTER_STRUCT_MAGIC_OFFSET = 8;
+  static final StatsCompressionCodec FOOTER_COMPRESSION_CODEC = StatsCompressionCodec.LZ4;
+
+  public static final String CREATED_BY_PROPERTY = "created-by";
+
+  static byte[] getMagic() {
+    return new byte[] {0x50, 0x46, 0x42, 0x31};
+  }
+
+  static void writeIntegerLittleEndian(OutputStream outputStream, int value) throws IOException {
+    outputStream.write(0xFF & value);
+    outputStream.write(0xFF & value >> 8);
```

Comment:
Maybe worth putting some parenthesis in for clarity

Reply:
good idea